### PR TITLE
Docs: mark up FTP.retrbinary and FTP.storbinary with param lists

### PR DESCRIPTION
**Pull request by erlend-aasland**
----
📚 Documentation preview 📚: https://cpython-previews--114399.org.readthedocs.build/

### CODE REVIEW

File: Doc/library/ftplib.rst [hunk 1]
```diff
@@ -201,14 +201,27 @@ FTP objects
...
-      each block of data received, with a single bytes argument giving the data
-      block. The optional *blocksize* argument specifies the maximum chunk size to
-      read on the low-level socket object created to do the actual transfer (which
-      will also be the largest size of the data blocks passed to *callback*).  A
-      reasonable default is chosen. *rest* means the same thing as in the
-      :meth:`transfercmd` method.
+      Retrieve a file in binary transfer mode.
+
+      :param str cmd:
+        An appropriate ``STOR`` command: :samp:`"STOR {filename}"`.
```
Comment (erlend-aasland) [inline Doc/library/ftplib.rst:207]: I'm not quite sure about the `:samp:` role here; perhaps the previous double backtick was good enough.

File: Doc/library/ftplib.rst [hunk 2]
```diff
@@ -201,14 +201,27 @@ FTP objects
...
+      :param int blocksize:
+         The maximum chunk size to read on the low-level
+         :class:`~socket.socket` object created to do the actual transfer.
+         This also corresponds to the largest size of data
+         that will be passed to *callback*.
+         Defaults to 8192.
+
+      :param int rest:
+         A ``REST`` command to be sent to the server.
+         See the corresponding named parameter to the :meth:`transfercmd` method.
```
Comment (erlend-aasland) [inline Doc/library/ftplib.rst]: The wording feels a little bit off-kilter. Perhaps a native speaker could help me improve this.

File: Doc/library/ftplib.rst [hunk 3]
```diff
@@ -232,16 +245,33 @@ FTP objects
 
    .. method:: FTP.storbinary(cmd, fp, blocksize=8192, callback=None, rest=None)
 
-      Store a file in binary transfer mode.  *cmd* should be an appropriate
-      ``STOR`` command: ``"STOR filename"``. *fp* is a :term:`file object`
-      (opened in binary mode) which is read until EOF using its :meth:`~io.IOBase.read`
```
Comment (erlend-aasland) [inline Doc/library/ftplib.rst:237]: FYI: `~io.IOBase.read` is a dead end.

File: Doc/library/ftplib.rst [hunk 4]
```diff
@@ -232,16 +245,33 @@ FTP objects
...
+         with its single argument being the data as :class:`bytes`.
+      :type callback: :term:`callable`
+
+      :param int rest:
+         A ``REST`` command to be sent to the server.
+         See the corresponding named parameter to the :meth:`transfercmd` method.
 
       .. versionchanged:: 3.2
-         *rest* parameter added.
+         The *rest* parameter added.
```
Comment (erlend-aasland) [inline Doc/library/ftplib.rst]:
```suggestion
         The *rest* parameter was added.
```

Comment (erlend-aasland) [reply Doc/library/ftplib.rst]: Resolved with 6497bb827d58ff01e79bc7e224248d0b945b0fbf

File: Doc/library/ftplib.rst (hunk 1 above)
Comment (ezio-melotti) [reply Doc/library/ftplib.rst:207]: Looks OK to me.

File: Doc/library/ftplib.rst [hunk 5]
```diff
@@ -201,14 +201,27 @@ FTP objects
...
+         for each block of data received,
+         with its single argument being the data as :class:`bytes`.
+      :type callback: :term:`callable`
+
+      :param int blocksize:
+         The maximum chunk size to read on the low-level
+         :class:`~socket.socket` object created to do the actual transfer.
+         This also corresponds to the largest size of data
+         that will be passed to *callback*.
+         Defaults to 8192.
```
Comment (ezio-melotti) [inline Doc/library/ftplib.rst]:
```suggestion
         Defaults to ``8192``.
```

File: Doc/library/ftplib.rst (hunk 2 above)
Comment (ezio-melotti) [inline Doc/library/ftplib.rst]: A couple of suggestions:
```suggestion
         See the documentation for the :meth:`transfercmd` method parameter
         with the same name.
```
```suggestion
         See the documentation for the *rest* parameter of the :meth:`transfercmd` method.
```

File: Doc/library/ftplib.rst [hunk 6]
```diff
@@ -232,16 +245,33 @@ FTP objects
...
+
+      :param fp:
+         A file object (opened in binary mode) which is read until EOF,
+         using its :meth:`~io.RawIOBase.read` method in blocks of size *blocksize*
+         to provide the data to be stored.
+      :type fp: :term:`file object`
+
+      :param int blocksize:
+         The read block size.
+         Defaults to 8192.
```
Comment (ezio-melotti) [inline Doc/library/ftplib.rst]:
```suggestion
         Defaults to ``8192``.
```

File: Doc/library/ftplib.rst [hunk 7]
```diff
@@ -232,16 +245,33 @@ FTP objects
...
+
+      :param callback:
+         A single parameter callable that is called
+         for each block of data sent,
+         with its single argument being the data as :class:`bytes`.
+      :type callback: :term:`callable`
+
+      :param int rest:
+         A ``REST`` command to be sent to the server.
+         See the corresponding named parameter to the :meth:`transfercmd` method.
```
Comment (ezio-melotti) [inline Doc/library/ftplib.rst]: A couple of suggestions:
```suggestion
         See the documentation for the :meth:`transfercmd` method parameter
         with the same name.
```
```suggestion
         See the documentation for the *rest* parameter of the :meth:`transfercmd` method.
```

File: Doc/library/ftplib.rst (hunk 5 above)
Comment (erlend-aasland) [reply Doc/library/ftplib.rst]: Is that the recommended way to mark up integer literals?

File: Doc/library/ftplib.rst (hunk 2 above)
Comment (erlend-aasland) [reply Doc/library/ftplib.rst]:
```suggestion
         See the documentation for the *rest* parameter of the :meth:`transfercmd` method
         for more details.
```

File: Doc/library/ftplib.rst (hunk 7 above)
Comment (erlend-aasland) [reply Doc/library/ftplib.rst]:
```suggestion
         See the documentation for the *rest* parameter of the :meth:`transfercmd` method
         for more details.
```

File: Doc/library/ftplib.rst (hunk 2 above)
Comment (erlend-aasland) [reply Doc/library/ftplib.rst]: ~~Alternatively, we can use substitutions here as well.~~ Let's leave that as a possible future enhancement.

**Comment by erlend-aasland** [comment]
Ezio's remarks have been addressed. I'm going to land this tomorrow.

File: Doc/tools/.nitignore [hunk 8]
```diff
@@ -45,7 +45,6 @@ Doc/library/enum.rst
 Doc/library/exceptions.rst
 Doc/library/faulthandler.rst
 Doc/library/fcntl.rst
-Doc/library/ftplib.rst
```
Comment (hugovk) [inline Doc/tools/.nitignore:48]: 🎉 🧹 📖 🎈

**Review by hugovk — APPROVED** [review]

**Pull request merged**